src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,404 @@
+
+            if (presetsContainer == null)
+                return;
+
+            separator.Visible = value;
+            presetsContainer.Visible = value;
+            addPresetButton.Visible = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        // Hide replaced native controls. Can't delete them because it will crash Godot.
+        var baseControl = GetChild(4);
+        baseControl.GetChild<Control>(4).Hide();
+        GetChild<Control>(5).Hide();
+        GetChild<Control>(6).Hide();
+        GetChild<Control>(7).Hide();
+
+        // Get controls
+        sliderROrH = baseControl.GetChild(0).GetChild<HSlider>(1);
+        sliderGOrS = baseControl.GetChild(1).GetChild<HSlider>(1);
+        sliderBOrV = baseControl.GetChild(2).GetChild<HSlider>(1);
+        sliderA = baseControl.GetChild(3).GetChild<HSlider>(1);
+        pickerButton = GetChild(1).GetChild<ToolButton>(1);
+        hsvCheckButton = GetNode<CheckButton>("MarginButtonsContainer/ButtonsContainer/HSVCheckButton");
+        rawCheckButton = GetNode<CheckButton>("MarginButtonsContainer/ButtonsContainer/RawCheckButton");
+        htmlColourEdit = GetNode<LineEdit>("MarginButtonsContainer/ButtonsContainer/HtmlColourEdit");
+        separator = GetNode<HSeparator>("Separator");
+        presetsContainer = GetNode<GridContainer>("PresetContainer");
+        addPresetButton = GetNode<TextureButton>("PresetButtonContainer/AddPresetButton");
+
+        // Update control state.
+        UpdateButtonsState();
+        PresetsEnabled = presetsEnabled;
+        PresetsVisible = presetsVisible;
+        OnColourChanged(Color);
+
+        // Load presets.
+        if (PresetsStorage.TryGetValue(PresetGroup, out groupStorage))
+        {
+            foreach (var colour in groupStorage.Colours)
+                GroupAddPreset(colour);
+        }
+        else
+        {
+            // Always ensure there is one so then we just modify it instead of having to check.
+            groupStorage = new PresetGroupStorage { Colours = GetPresets().ToList() };
+            PresetsStorage.Add(PresetGroup, groupStorage);
+        }
+
+        // Add current preset handlers to preset group
+        groupStorage.AddPreset += GroupAddPreset;
+        groupStorage.ErasePreset += GroupErasePreset;
+
+        UpdateTooltips();
+    }
+
+    public override void _ExitTree()
+    {
+        groupStorage.AddPreset -= GroupAddPreset;
+        groupStorage.ErasePreset -= GroupErasePreset;
+        base._ExitTree();
+    }
+
+    public override void _Notification(int what)
+    {
+        if (what == NotificationTranslationChanged)
+            UpdateTooltips();
+
+        base._Notification(what);
+    }
+
+    public new void AddPreset(Color colour)
+    {
+        if (groupStorage.Colours.Contains(colour))
+            return;
+
+        groupStorage.Colours.Add(colour);
+
+        // Broadcast to all group numbers.
+        groupStorage.AddPreset(colour);
```

Comment:
Instead of having these be separate. I think there should be one method in groupStorage that does this both. That way someone doesn't modify Colours accidentally without also broadcasting the change. I'll comment more in that class.